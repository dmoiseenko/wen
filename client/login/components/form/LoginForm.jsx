import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

import Input from '../../../common/components/input/Input';
import { emailField, passwordField } from '../../core/form.constants';


export default function LoginForm({ signIn }) {
  function handleClick(event) {
    event.preventDefault();
    signIn();
  }

  return (
    <form className="login-form">
      <Field
        name={emailField}
        component={Input}
        placeholder="mail@example.com"
        type="text"
        inputClassName="email-input"
        iconClassName="fa fa-user"
      />
      <Field
        name={passwordField}
        component={Input}
        placeholder="●●●●●●●"
        type="password"
        inputClassName="password-input"
        iconClassName="fa fa-lock"
      />
      <p className="control login">
        <button
          type="submit"
          onClick={handleClick}
          className="login-button"
        >
          Login
        </button>
      </p>
    </form>
  );
}

LoginForm.propTypes = {
  signIn: PropTypes.func.isRequired
};
