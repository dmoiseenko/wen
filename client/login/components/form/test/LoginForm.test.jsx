import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from '../LoginForm';


describe('<LoginForm />', () => {
  const props = {
    signIn: jest.fn()
  };

  it('should render', () => {
    const wrapper = shallow(
      <LoginForm {...props} />
    );

    expect(wrapper.node).toMatchSnapshot();
  });

  it('should prevent default event action and dispatch signIn when click fires', () => {
    const signIn = jest.fn();
    const event = {
      preventDefault: jest.fn()
    };

    const wrapper = shallow(
      <LoginForm
        signIn={signIn}
      />
    );

    wrapper.find('.login-button').simulate('click', event);

    expect(signIn).toHaveBeenCalled();
    expect(event.preventDefault).toHaveBeenCalled();
  });
});
