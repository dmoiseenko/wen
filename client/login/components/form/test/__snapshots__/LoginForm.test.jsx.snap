// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render 1`] = `
<form
  className="login-form"
>
  <Field
    component={[Function]}
    iconClassName="fa fa-user"
    inputClassName="email-input"
    name="emailField"
    placeholder="mail@example.com"
    type="text"
  />
  <Field
    component={[Function]}
    iconClassName="fa fa-lock"
    inputClassName="password-input"
    name="passwordField"
    placeholder="●●●●●●●"
    type="password"
  />
  <p
    className="control login"
  >
    <button
      className="login-button"
      onClick={[Function]}
      type="submit"
    >
      Login
    </button>
  </p>
</form>
`;
